### PR TITLE
Fixes cyborg initial chat

### LINKED ISSUE #29774: Freed prototype cyborgs not recruitable?
**Describe the bug**  
Prototype cyborgs fresh out of the AutoDoc are not properly recruitable. 

**To Reproduce**  
1. Find an AutoDoc.
2. Spawn an inactive prototype cyborg with cheats and drop it on the autodoc.
3. Remove its Personality Override chip with the Autodoc.
4. Attempt to recruit it.

**Expected behavior**  
Prototype cyborgs freed in this manner 

**Versions and configuration(please complete the following information):**
 - Game Version: 0.D-1789-gc0d27c5
 - Graphics version: Tiles

**Additional context**  
Might have something to do with their attitude being set to Ignoring? Which is not an attitude I see on NPCs a lot.

### PR DESCRIPTION
**Pull request by GearheadLydia**
Fixes #29774 and makes freed cyborgs actually recruitable. Also throws in a nice little opinion bonus.

#### Summary
Summary: Content "Fixed cyborg initial recruitment dialogue"

#### Purpose of change
Fixes #29774 and allows cyborgs to be recruited.

#### Additional context
Right now, this recruitment dialogue is a little unsubtle. It may be worth looking into putting together a randomized initial opinion modifier that controls whether or not the cyborg will offer to join the player. I'll look into doing this if there is interest.

### CODE REVIEW

**Comment by mlangsdorf** [comment]
Recruitment is almost always a PERSUADE or INTIMIDATE trial.

**Comment by GearheadLydia** [comment]
Right. I'm thinking the only reason it wasn't in this case is because it was considered hard enough to get to the point where recruitment is possible. I'll see what I can do.

**Comment by GearheadLydia** [comment]
Too complicated?
`[
  {
    "id": "TALK_CYBORG_1",
    "type": "talk_topic",
    "dynamic_line": "I... I'm free. *Zzzt* I'm actually free! *bzzz* Look, you're the first person I've seen in a long time.",
  
    "responses": [
      {
        "switch": true,
        "default": true,
        "text": "That's because I saved you. Listen, I could use your help...",
        "trial": { "type": "PERSUADE", "difficulty": 0 },
        "success": { "topic": "TALK_CYBORG_FRIENDLY", "opinion": { "trust": 1, "value": 1 } },
        "failure": { "topic": "TALK_CYBORG_WARY", "opinion": { "anger": 1, "fear": 1 } }
      },
      {
        "switch": true,
        "default": true,
        "text": "Big fucking deal. Do what I say or I'll shove that chip back where it belongs.",
        "trial": { "type": "INTIMIDATE", "difficulty": 20, "mod": [ [ "AGGRESSION", -2 ], [ "BRAVERY", -2 ] ] },
        "success": { "topic": "TALK_CYBORG_FEARFUL", "opinion": { "trust": -4, "fear": 3, "value": -1, "anger": 4 } },
        "failure": { "topic": "TALK_CYBORG_BREAKDOWN", "opinion": { "trust": -4, "value": -5, "anger": 10 } }
      },
      {
        "switch": true,
        "default": true,
        "text": "Sorry, I'm nobody. Enjoy your freedom, I guess.",
        "topic": "TALK_DONE"
      }
    ]
  },
  {
    "id": "TALK_CYBORG_FRIENDLY",
	"type": "talk_topic",
	"dynamic_line":
	{
        "npc_has_effect": "asked_to_follow",
        "no": "*buzz* Great! So what happens now?",
        "yes": "Well, that's good to hear. Still not coming with you, though."
	},
    "responses": [
      {
        "switch": true,
        "condition": { "npc_has_effect": "asked_to_follow" },
        "text": "Right, right. I'll ask later.",
        "topic": "TALK_DONE"
      },
      {
        "switch": true,
        "default": true,
        "text": "Come with me. We can help each other out.",
        "trial": { "type": "PERSUADE", "difficulty": 0, "mod": [ [ "value", 2 ] ]},
        "success": { "topic": "TALK_AGREE_FOLLOW", "effect": "follow", "opinion": { "trust": 1, "value": 1 } },
        "failure": { "topic": "TALK_DENY_FOLLOW", "effect": "deny_follow", "opinion": { "trust": -1, "fear": 1 } }
      },
      {
        "switch": true,
        "default": true,
        "text": "We both go our separate ways. Enjoy your freedom.",
        "topic": "TALK_DONE"
      }
	]
  },
  {
    "id": "TALK_CYBORG_WARY",
    "type": "talk_topic",
    "dynamic_line": "...Wait. *BEEP* Why do I believe you? *ZZZT* You could be just as bad as them!",
  
    "responses": [
      {
        "switch": true,
        "default": true,
        "text": "I wouldn't have pulled your chip out if I didn't want you to think for yourself.",
        "trial": { "type": "PERSUADE", "difficulty": 5, "mod": [ [ "TRUST", 2 ], [ "VALUE", 2 ] ]},
        "success": { "topic": "TALK_CYBORG_FRIENDLY", "opinion": { "trust": 1, "value": 1 } },
        "failure": { "topic": "TALK_CYBORG_LEAVES" }
      },
      {
        "switch": true,
        "default": true,
        "text": "For all you know, I did. I'm being nice for now. You'd better hope that it lasts.",
        "trial": { "type": "INTIMIDATE", "difficulty": 20, "mod": [ [ "AGGRESSION", -2 ], [ "BRAVERY", -2 ] ] },
        "success": { "topic": "TALK_CYBORG_FEARFUL", "opinion": { "trust": -4, "fear": 3, "value": -1, "anger": 4 } },
        "failure": { "topic": "TALK_CYBORG_BREAKDOWN", "opinion": { "trust": -4, "value": -5, "anger": 10 } }
      },
      {
        "switch": true,
        "default": true,
        "text": "Forget it. Enjoy your freedom.",
        "topic": "TALK_DONE"
      }
    ]
  },
  {
    "id": "TALK_CYBORG_FEARFUL",
	"type": "talk_topic",
	"dynamic_line": "Okay, okay, *BUZZ* I'm sorry! Don't hurt me again! Anything but the chip!",
	"responses": [
      {
        "switch": true,
        "default": true,
        "text": "Follow me and do my bidding, then.",
        "trial": { "type": "INTIMIDATE", "difficulty": 20, "mod": [ [ "FEAR", 8 ], [ "VALUE", 2 ], [ "TRUST", 2 ], [ "BRAVERY", -2 ] ] },
        "success": { "topic": "TALK_AGREE_FOLLOW", "effect": "follow", "opinion": { "trust": -4, "fear": 3, "value": -1, "anger": 4 } },
        "failure": { "topic": "TALK_CYBORG_BREAKDOWN", "opinion": { "trust": 4, "value": -5, "anger": 10 } }
      },
      {
        "switch": true,
        "default": true,
        "text": "Get out of my sight.",
        "topic": "TALK_DONE",
        "effect": "flee",
        "opinion": { "trust": -1, "fear": -2, "anger": -2 }
      },
      {
        "switch": true,
        "default": true,
        "text": "No, *I'm* sorry, I didn't mean that. Go do what you want.",
        "trial": { "type": "PERSUADE", "difficulty": 10 },
        "success": { "topic": "TALK_DONE", "opinion": { "trust": 1, "value": 1, "anger": -1 } },
        "failure": { "topic": "TALK_CYBORG_BREAKDOWN", "opinion": { "trust": 4, "value": -5, "anger": 10 } }
      }
	]
  },
  {
    "id": "TALK_CYBORG_BREAKDOWN",
	"type": "talk_topic",
	"dynamic_line": "...kill... *ZTZTZT* ...you!",
	"responses": [
      {
        "switch": true,
        "default": true,
        "text": "Run while you still can!",
        "trial": { "type": "INTIMIDATE", "difficulty": 20, "mod": [ [ "FEAR", 8 ], [ "VALUE", 2 ], [ "TRUST", 2 ], [ "BRAVERY", -2 ] ] },
        "success": { "topic": "TALK_DONE", "effect": "flee" },
        "failure": { "topic": "TALK_DONE", "effect": "hostile" }
      },
      {
        "switch": true,
        "default": true,
        "text": "Not if I kill you first!",
        "topic": "TALK_DONE",
        "effect": "hostile"
      }
	]
  },
  {
	"id": "TALK_CYBORG_LEAVES",
	"type": "talk_topic",
	"dynamic_line": "Yeah, *beeeeep* I don't think so. Bye.",
	"responses": [
	  {
	    "text": "Suit yourself.",
		"topic": "TALK_DONE"
	  }
	]
  }
]`

**Comment by Fris0uman** [comment]
Thanks for fixing it. The idea of making it automatic was that you saved them and now they're thankfull and the reward is a follower.
I think it would feel weird if they could just tell you to buzz off and then leave.

**Comment by GearheadLydia** [comment]
Someone take a look at what it's like now? I have no idea if it's balanced, but at least it seems to work.

File: data/json/npcs/TALK_CYBORG_1.json [hunk 1]
```diff
@@ -2,10 +2,169 @@
...
+      }
     ]
+  },
+  {
+    "id": "TALK_CYBORG_FRIENDLY",
+	"type": "talk_topic",
+	"dynamic_line": "*buzz* Great! So what happens now?",
+    "responses": [
+      {
+        "switch": true,
```
Comment (mlangsdorf) [inline data/json/npcs/TALK_CYBORG_1.json]: you don't need a switch and a default in both response objects.  If you're not intending to short circuit the process, you don't need a switch.

File: data/json/npcs/TALK_CYBORG_1.json [hunk 2]
```diff
@@ -2,10 +2,169 @@
...
+	]
+  },
+  {
+    "id": "TALK_CYBORG_WARY",
+    "type": "talk_topic",
+    "dynamic_line": "...Wait. *BEEP* Why do I believe you? *ZZZT* You could be just as bad as them!",
+  
+    "responses": [
+      {
+        "switch": true,
```
Comment (mlangsdorf) [inline data/json/npcs/TALK_CYBORG_1.json]: same comment on these.

File: data/json/npcs/TALK_CYBORG_1.json [hunk 3]
```diff
@@ -2,10 +2,169 @@
...
+	"dynamic_line":
+	{
+        "npc_has_var": "cyborg_has_talked",
+		"type": "dialogue",
+		"context": "cyborg",
+		"value": "yes",
+		"no": "I... I'm free. *Zzzt* I'm actually free! *bzzz* Look, you're the first person I've seen in a long time.",
+		"yes": "Hey again. *kzzz*"
+	},
+
```
Comment (mlangsdorf) [inline data/json/npcs/TALK_CYBORG_1.json]: Consider adding:
```
    "speaker_effect": {
      "effect": { "npc_add_var": "cyborg_has_talked", "type": "dialogue", "context": "cyborg", "value": "yes" },
      "sentinel": "cyborg_has_talked"
    },
```
here, which will 100% of the time set the cyborg_has_talked var the first time you talk to the cyborg, and then remove all the npc_add_var effects afterwards.

Comment (GearheadLydia) [reply data/json/npcs/TALK_CYBORG_1.json]: I was looking at doing it that way, but I wasn't 100% sure how it worked, so I played it safe.

**Comment by Fris0uman** [comment]
That looks pretty good . The only thing is that  (for now) putting back the chip is impossible as it's always destroyed by the autodoc (some last resort failsafe to keep it to get out of the labs maybe). Of course the cyborgs don't necessarily know that...

**Comment by GearheadLydia** [comment]
> That looks pretty good . The only thing is that (for now) putting back the chip is impossible as it's always destroyed by the autodoc (some last resort failsafe to keep it to get out of the labs maybe). Of course the cyborgs don't necessarily know that...

My intention was that it's supposed to be an empty, unnecessarily cruel threat. I don't really play the game that way, but I wanted to add some variety.

File: data/json/npcs/TALK_CYBORG_1.json [hunk 4]
```diff
@@ -2,10 +2,150 @@
...
+	"dynamic_line": "...kill... *ZTZTZT* ...you!",
+	"responses": [
+      {
+        "text": "Run while you still can!",
+        "trial": { "type": "INTIMIDATE", "difficulty": 20, "mod": [ [ "FEAR", 8 ], [ "VALUE", 2 ], [ "TRUST", 2 ], [ "BRAVERY", -2 ] ] },
+        "success": { "topic": "TALK_DONE", "effect": "flee" },
+        "failure": { "topic": "TALK_DONE", "effect": "hostile" }
+      },
+      {
+        "text": "Not if I kill you first!",
```
Comment (mlangsdorf) [inline data/json/npcs/TALK_CYBORG_1.json]: Linter wants this entire object on one line for some reason.  Obey the linter.
```
{ "text": "Not if I kill you first!", "topic": "TALK_DONE", "effect": "hostile" },
```

File: data/json/npcs/TALK_CYBORG_1.json [hunk 5]
```diff
@@ -2,10 +2,150 @@
...
+        "success": { "topic": "TALK_DONE", "opinion": { "trust": 1, "value": 1, "anger": -1 } },
+        "failure": { "topic": "TALK_DONE", "effect": "hostile" }
+      }
+	]
+  },
+  {
+	"id": "TALK_CYBORG_LEAVES",
+	"type": "talk_topic",
+	"dynamic_line": "Yeah, *beeeeep* I don't think so. Bye.",
+	"responses": [
```
Comment (mlangsdorf) [inline data/json/npcs/TALK_CYBORG_1.json]:
```suggestion
	"responses": [ { "text": "Suit yourself.", "topic": "TALK_DONE" } ]
```
Again, the linter must be appeased.

File: data/json/npcs/TALK_CYBORG_1.json [hunk 6]
```diff
@@ -2,10 +2,150 @@
...
+        "success": { "topic": "TALK_CYBORG_FRIENDLY", "opinion": { "trust": 1, "value": 1 } },
+        "failure": { "topic": "TALK_CYBORG_LEAVES" }
+      },
+      {
+        "text": "For all you know, I did. I'm being nice for now. You'd better hope that it lasts.",
+        "trial": { "type": "INTIMIDATE", "difficulty": 20, "mod": [ [ "AGGRESSION", -2 ], [ "BRAVERY", -2 ] ] },
+        "success": { "topic": "TALK_CYBORG_FEARFUL", "opinion": { "trust": -4, "fear": 3, "value": -1, "anger": 4 } },
+        "failure": { "topic": "TALK_CYBORG_BREAKDOWN", "opinion": { "trust": -4, "value": -5, "anger": 10 } }
+      },
+      {
```
Comment (mlangsdorf) [inline data/json/npcs/TALK_CYBORG_1.json]:
```suggestion
      { "text": "Forget it. Enjoy your freedom.", "topic": "TALK_DONE" }
```

File: data/json/npcs/TALK_CYBORG_1.json [hunk 7]
```diff
@@ -2,10 +2,150 @@
...
+	"type": "talk_topic",
+	"dynamic_line": "*buzz* Great! So what happens now?",
+    "responses": [
+      {
+        "text": "Come with me. We can help each other out.",
+        "trial": { "type": "PERSUADE", "difficulty": 0, "mod": [ [ "value", 2 ] ]},
+        "success": { "topic": "TALK_AGREE_FOLLOW", "effect": "follow", "opinion": { "trust": 1, "value": 1 } },
+        "failure": { "topic": "TALK_DENY_FOLLOW", "effect": "deny_follow", "opinion": { "trust": -1, "fear": 1 } }
+      },
+      {
```
Comment (mlangsdorf) [inline data/json/npcs/TALK_CYBORG_1.json]:
```suggestion
      { "text": "We both go our separate ways. Enjoy your freedom.", "topic": "TALK_DONE" }
```

**Comment by GearheadLydia** [comment]
Thanks. How's that?

**Pull request merged**